src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1352,10 +1351,8 @@ emit_vector_create_elementwise (
 	if (vector_size == 16) {
 		for (int i = 0; i < param_count; ++i) {
 			if (!is_const (args[i])) {
-				all_const = false;
-				break;
```

Comment:
This was the remaining issue.

Mono only supports 128-bit `xconst` today, but we're still trying to handle the case where only some inputs are constant; such as `Vector128.Create(1, 2, x, 4)`. This was early exiting in the case that not all inputs were constant so in the main loop below we ended up getting `Vector128.Create(1, 2, x, 0)` instead.

Fixed it to not early exit and only skip parameters if some were constant; which lets us get the correct value instead.